src/gh-webhook/github.ts

```diff
@@ -0,0 +1,103 @@
+import crypto from 'node:crypto';
+import type { Env } from '..';
+
+export async function handleGitHubWebhook(request: Request, env: Env): Promise<Response> {
+  if (request.method !== 'POST') {
+    return new Response('Method not allowed', { status: 405, statusText: 'Method Not Allowed' });
+  }
+
+  const githubSecret = process.env.GITHUB_SECRET;
+  const WebhookURL = process.env.WEBHOOK_URL;
+
+  if (!githubSecret || !WebhookURL) {
+    return new Response('Internal server error', { status: 500, statusText: 'Internal Server Error' });
+  }
+
+  const authorized = await isAuthorized(request, githubSecret);
+
+  if (!authorized) {
+    return new Response('Unauthorized', { status: 401, statusText: 'Unauthorized' });
+  }
+
+  const isHuman = await isHumanEvent(request);
+
+  if (!isHuman) {
+    return new Response('Event skipped', { status: 200, statusText: 'OK' });
+  }
+
+  const sent = await sendToWebhook(request);
+
+  if (!sent) {
+    return new Response('Failed to send to Discord', { status: 500, statusText: 'Internal Server Error' });
+  }
+
+  return new Response('Event processed', { status: 200, statusText: 'OK' });
+}
+
+async function isAuthorized(request: Request, githubSecret: string): Promise<boolean> {
+  const header_signature = request.headers.get('x-hub-signature-256');
```

Comment:
use camelCase for variable naming, i should probably enable the lint rule for that :P

Reply:
Yea I wonder which tool could do something like that :D